legend-engine-xt-serviceStore-executionPlan/src/test/java/org/finos/legend/engine/plan/execution/stores/service/securitySchemes/TestApiKeyAuthenticationWithApiKeySecurityScheme.java

```diff
@@ -0,0 +1,99 @@
+
+public class TestApiKeyAuthenticationWithApiKeySecurityScheme extends ServiceStoreTestSuite
+{
+    private static String pureGrammar;
+
+    @BeforeClass
+    public static void setup()
+    {
+        setupServer("securitySchemes");
+
+        String serviceStore =
+                "###ServiceStore\n" +
+                        "ServiceStore meta::external::store::service::showcase::store::TradeProductServiceStore\n" +
+                        "(\n" +
+                        "   description : 'Showcase Service Store';\n" +
+                        "   securitySchemes : {\n" +
+                        "       api1 : ApiKey\n" +
+                        "               {\n" +
+                        "                   location : 'cookie';\n" +
+                        "                   keyName : 'apiKey1';\n" +
+                        "               }\n" +
+                        "   };\n" +
+                        "   ServiceGroup TradeServices\n" +
+                        "   (\n" +
+                        "      path : '/trades';\n" +
+                        "\n" +
+                        "      Service AllTradeService\n" +
+                        "            (\n" +
+                        "               path : '/allTradesService2';\n" +
+                        "               method : GET;\n" +
+                        "               security : [api1];\n" +
+                        "               response : [meta::external::store::service::showcase::domain::S_Trade <- meta::external::store::service::showcase::store::tradeServiceStoreSchemaBinding];\n" +
+                        "            )\n" +
+                        "   )  \n" +
+                        ")";
+
+        String serviceStoreConnection = "###Connection\n" +
+                "ServiceStoreConnection meta::external::store::service::showcase::connection::serviceStoreConnection\n" +
+                "{\n" +
+                "    store   : meta::external::store::service::showcase::store::TradeProductServiceStore;\n" +
+                "    baseUrl : 'http://127.0.0.1:port';\n" +
+                "    auth    : {\n" +
+                "                 api1  : ApiKey\n" +
+                "                 {\n" +
+                "                    location : 'header';\n" +
+                "                    keyName : 'key1';\n" +
+                "                    value : SystemPropertiesSecret\n" +
+                "                    {\n" +
+                "                        systemPropertyName : 'property1';\n" +
+                "                    }\n" +
+                "                 }\n" +
+                "            };\n" +
+                "}";
+
+        pureGrammar = serviceStore + "\n\n" + serviceStoreConnection.replace("port", String.valueOf(getPort())) + "\n\n" + ServiceStoreTestUtils.readGrammarFromPureFile("/securitySchemes/testGrammar.pure");
+    }
+
+    @Test
+    public void testAuthentication()
```

Comment:
better name please, also lets have test for diff location/auth specs combination

Reply:
We just have support for cookie location initially.